feat(qcon24): First version of post (#445)
BedrockStreaming/tech.bedrockstreaming.com: _posts/2024-04-17-qcon-london-2024.md

---
layout: post
title: QCon London 2024
description: A few notes and ideas, coming back from this QCon conference at London.
author: [c_petit, j_foray, v_gallissot]
tags: [cloud, devops, conference, 2024]
color: rgb(251,87,66)
thumbnail: "/images/posts/2024-04-17-qcon-london-2024/2024-04-09-Qcon-London-arrival.jpg"
---

> We really enjoyed this [QCon conference in London](https://qconlondon.com/). We really enjoyed the QCon conference in London. It's based on 5 tracks a day, covering resilience, scalability, architecture, monitoring, performance, management and all the subjects that speak to tech companies. None of these tracks are sponsored: they're all about feedback, with no marketing whatsoever. There is a 6th track where sponsorship is permitted, and it's the only one. There aren't many sponsorship stands, and you go to this conference to talk to your peers first and foremost.

Reviewer: duplicate phrase.

Author: You're right, I fixed :wink:

Several presenters spoke of asynchronous and 'real-time' scalability (meaning millions of users in a matter of seconds). The bottom line for all of them is that there is no way of scaling the necessary resources so quickly: everyone pre-scales. The pre-scaling itself can take a long time, so for a particular event (such as a sports match), resources need to be launched several hours in advance. Asynchronous processing can help to spread out all the non-critical calculations over time.

Don’t forget to be patient ! It really seems to go against the tide, yet designing long-time running APIs (meaning APIs that can handle waiting) really seems to be the key of success when trying to address scalability issues. Likewise try as much as possible not to overlap your APIs field of expertise.

On the organisational level, architects of all kinds must identify or create links between all elements in the ecosystem.

The architect's vision is to understand the choices and make compromises between the teams so that the whole is coherent. There is no no lock-in. Every choice brings both positives and negatives.

## Sustainability

> A big scary or fancy world, however it is our responsibility to deal with, and know what, it turns out this part often rimes with reducing costs. 

Reviewer 2: ```suggestion
Sustainability is a bit scary and fancy word, however it is our responsibility to deal with it... And know what? It turns out this often rimes with reducing costs! 
```

Author: I've taken the whole sentence back, it sounds better now

The aim is to use energy when it is really needed: switching off dev and staging environments when employees are resting. Limit over-provisioning of resources and pre-scaling times.
All cloud providers include a sustainability section in their 'well-architected' programs, a section that is independent of performance and costs, even though it provides co-benefits with those two subjects.

It really is in companies' interests to move in this direction.

## Management, People and  Process

Maybe first keep in mind that decisions are not right or bad, they just are taken by a person or a quorum based on the context they had on a given time. And let’s face it, your organisation will need to change over time especially if you begin as a start-up and now need to scale-up (oh an interesting topic of scaling coming back!).

> As for scaling your infrastructure, you need to realise you will need to make compromises. For example, Trainline shared with us that their three pillars are: Alignment, Productivity and Code Quality. You can't perform in the three at the same time, you need to choose which one(s) to let go a little in order to achieve the company's goal. Importantly what is working at a given time in a certain situation could not be appropriate anymore months later.

Reviewer 2: ```suggestion
As for scaling your infrastructure, you need to realise that you will need to make compromises. For example, Trainline shared with us that their three pillars are: Alignment, Productivity and Code Quality. You can't perform in the three at the same time, you need to choose which one(s) to let go a little in order to achieve the company's goal. Importantly what is working at a given time in a certain situation could not be appropriate anymore months later.
```

## Platform Engineering

The last day of the conference offered a dedicated track on platform engineering, where we listened to the insights shared by the speakers. As they delved into their own experiences and learnings, we found striking parallels with our own journey at Bedrock over the past few years.

![Lessons from Trainline](/images/posts/2024-04-17-qcon-london-2024/2024-04-09-Trainline-lessons.jpg)

> One of the most resonant themes was the importance of treating the platform as a product—a perspective shift that has significantly influenced our approach. By emphasising the need for a smoother experience that balances consistency, performance, security, and development speed, the speakers reinforced the notion that our platform isn't just a set of tools; it's a crucial enabler for our developers' success.

Reviewer 2: > treating the platform as a product—a perspective shift that has significantly influenced our approach

I don't understand this sentence... maybe there's an issue around `product—a`?

Author: I guess it was a style effect that wasn't well copy/pasted :shrug:

The speakers also shared their experiences with empowering developers to take ownership of certain tasks, such as fixing issues surfaced by conformity checks and contributing features that solved their own challenges back up to the general platform to benefit the whole organisation. This aligns closely with our own DevOps principles, emphasising collaboration and shared responsibility across teams.

As we reflect on the insights shared by the conference speakers, we're reminded of the importance of continuous improvement and iteration in our platform engineering efforts. By learning from our mistakes, embracing new perspectives, and staying attuned to the needs of our users, we can continue to evolve our platform into a truly indispensable resource for our development community.

## Conclusion

It was our first QCon for most of us. We appreciated the transparency of the speakers and the commitment of the organisers, who made the event a success on every level: the intensity of the conferences, the quality of the talks, the really good and varied food and the discreet presence of the sponsors.

> Without being very technical, this conference is always a source of inspiration for us, as it focuses on feedback and encourages discussion with our peers.

Reviewer 2: ```suggestion
Without being very technical, this conference is a source of inspiration for us, as it focuses on feedback and encourages discussion with our peers.
```

(because using `always` here goes against `It was our first QCon for most of us` above)
